cdap-app-templates/cdap-etl/cdap-etl-proto/src/main/java/io/cdap/cdap/etl/proto/v2/spec/StageSpec.java

```diff
@@ -53,6 +56,7 @@ private StageSpec(String name, PluginSpec plugin, Map<String, Schema> inputSchem
     this.name = name;
     this.plugin = plugin;
     this.inputSchemas = Collections.unmodifiableMap(inputSchemas);
+    this.inputStages = Lists.newArrayList(inputSchemas.keySet());
```

Comment:
better use `Collections.unmodifiableList()`

Reply:
this code has been removed, but in the [fallback path ](https://github.com/cdapio/cdap/pull/11948#discussion_r396692540), I'm using `ImmutableList.copyOf`.